x/stakeibc/keeper/unbonding_records_test.go

```diff
@@ -117,3 +118,185 @@ func (s *KeeperTestSuite) TestSubmitHostZoneUnbondingMsg_ErrorSubmittingUnbondin
+					UnbondingTime:     unbondingTime,
+				},
+				{
+					HostZoneId:        OsmoChainId,
+					NativeTokenAmount: sdk.NewInt(2_000_000),
+					Status:            recordtypes.HostZoneUnbonding_EXIT_TRANSFER_QUEUE,
+					UnbondingTime:     unbondingTime,
+				},
+			},
+		},
+		{
+			EpochNumber: 2,
+			HostZoneUnbondings: []*recordtypes.HostZoneUnbonding{
+				{
+					HostZoneId:        HostChainId,
+					NativeTokenAmount: sdk.NewInt(5_000_000),
+					Status:            recordtypes.HostZoneUnbonding_CLAIMABLE,
+				},
+				{
+					HostZoneId:        OsmoChainId,
+					NativeTokenAmount: sdk.NewInt(5_000_000),
+					Status:            recordtypes.HostZoneUnbonding_UNBONDING_QUEUE,
+				},
+			},
+		},
+	}
+	for _, epochUnbondingRecord := range epochUnbondingRecords {
+		s.App.RecordsKeeper.SetEpochUnbondingRecord(s.Ctx, epochUnbondingRecord)
+	}
+
+	for _, hostZone := range hostZones {
+		s.App.StakeibcKeeper.SetHostZone(s.Ctx, hostZone)
+	}
+	s.App.StakeibcKeeper.SetEpochTracker(s.Ctx, dayEpochTracker)
+
+	return SweepUnbondedTokensTestCase{
+		epochUnbondingRecords: epochUnbondingRecords,
+		hostZones:             hostZones,
+		lightClientTime:       lightClientTime,
+	}
+}
+
+func (s *KeeperTestSuite) TestSweepAllUnbondedTokensForHostZone_success() {
+	tc := s.SetupSweepAllUnbondedTokensForHostZone()
+	success, sweepAmount := s.App.StakeibcKeeper.SweepAllUnbondedTokensForHostZone(s.Ctx, tc.hostZones[0], tc.epochUnbondingRecords)
+	s.Require().True(success, "sweep all tokens for hostzone GAIA success")
+	s.Require().Equal(sdk.NewInt(2_000_000), sweepAmount, "sweep all unbonded tokens (with status EXIT_TRANSFER_QUEUE) for hostone GAIA success")
+	success, sweepAmount = s.App.StakeibcKeeper.SweepAllUnbondedTokensForHostZone(s.Ctx, tc.hostZones[1], tc.epochUnbondingRecords)
+	s.Require().True(success, "sweep all tokens for hostzone OSMO success")
+	s.Require().Equal(sdk.NewInt(3_000_000), sweepAmount, "sweep all unbonded tokens (with status EXIT_TRANSFER_QUEUE) for hostone OSMO success")
+}
+
+func (s *KeeperTestSuite) TestSweepAllUnbondedTokensForHostZone_HostZoneUnbondingNotFound() {
+	tc := s.SetupSweepAllUnbondedTokensForHostZone()
+	epochUnbondingRecords := s.App.RecordsKeeper.GetAllEpochUnbondingRecord(s.Ctx)
+	for _, epochUnbonding := range epochUnbondingRecords {
+		epochUnbonding.HostZoneUnbondings = []*recordtypes.HostZoneUnbonding{
+			epochUnbonding.HostZoneUnbondings[0],
+		}
+		s.App.RecordsKeeper.SetEpochUnbondingRecord(s.Ctx, epochUnbonding)
+	}
+	success, sweepAmount := s.App.StakeibcKeeper.SweepAllUnbondedTokensForHostZone(s.Ctx, tc.hostZones[1], tc.epochUnbondingRecords)
+	s.Require().True(success, "sweep all tokens for hostzone OSMO still success (even when overflow happended)")
+	s.Require().Equal(sdk.NewInt(0), sweepAmount, "No Unbonded tokens for hostzone OSMO is sweeped (because we removed the hostzone earlier)")
+}
+
+func (s *KeeperTestSuite) TestSweepAllUnbondedTokensForHostZone_blockTimeForHostZoneNotFound() {
```

Comment:
I don't think this is the case for testing because it's just the log in this function @hoangdv2429

Reply:
yes, I removed redundant tests with new commit.